### PR TITLE
feat: handle new nakamoto wire format

### PR DESCRIPTION
**Pull request by hugocaillard**

### CODE REVIEW

File: components/clarity-repl/src/codec/mod.rs [hunk 1]
```diff
@@ -2451,7 +2571,36 @@ impl StacksMessageCodec for TransactionPayload {
...
+                    }
+                } else {
+                    return Err(CodecError::DeserializeError("Failed to parse nakamoto coinbase transaction -- did not receive an optional recipient principal value".to_string()));
+                };
+                let vrf_proof_bytes: Vec<u8> = read_next(fd)?;
+                let Some(vrf_proof) = VRFProof::from_bytes(&vrf_proof_bytes) else {
+                    return Err(CodecError::DeserializeError(
+                        "Failed to decode coinbase VRF proof".to_string(),
+                    ));
+                };
```
Comment (janniks) [inline components/clarity-repl/src/codec/mod.rs:2597]:
```suggestion
                let vrf_proof: VRFProof = read_next(fd)?;
```
stacks-core recently simplified this, maybe doesn't matter though

**Review by janniks — CHANGES_REQUESTED** [review]
Heads up this PR will change the TenureChange again a bit: https://github.com/stacks-network/stacks-core/pull/4114
Other than that it looks good 👍 lmk if you need to merge a version prior to stacks-core finalisation.

File: components/clarity-repl/src/codec/mod.rs (hunk 1 above)
Comment (hugocaillard) [reply components/clarity-repl/src/codec/mod.rs:2597]: Ah cool stuff, I was looking at our own clarity-wasm-next branch, thanks @janniks!

**Comment by hugocaillard** [comment]
@janniks I converted to a draft while waiting for the stacks-core PR to be merged

**Comment by hugocaillard** [comment]
Cherry picked in #1266

**Pull request closed without merging**